src/hooks/useResultCallback.ts

```diff
@@ -1,19 +1,104 @@
+import { pipe } from "@effect/data/Function"
+import * as Hash from "@effect/data/Hash"
 import * as Effect from "@effect/io/Effect"
-import * as Exit from "@effect/io/Exit"
-import * as Fiber from "@effect/io/Fiber"
+import type * as Fiber from "@effect/io/Fiber"
 import * as Ref from "@effect/io/Ref"
 import * as Runtime from "@effect/io/Runtime"
 import * as Stream from "@effect/stream/Stream"
 import type { ResultBag } from "effect-react/hooks/useResultBag"
-import { updateNext, useResultBag } from "effect-react/hooks/useResultBag"
+import { initialTrackedProps, makeResultBag, updateTrackedProps } from "effect-react/hooks/useResultBag"
 import type { RuntimeContext } from "effect-react/internal/runtimeContext"
 import * as Result from "effect-react/Result"
-import { useCallback, useContext, useEffect, useRef, useState } from "react"
+import type { DependencyList } from "react"
+import { useCallback, useContext, useRef, useSyncExternalStore } from "react"
 
-type FiberState<E> = { readonly _tag: "Idle" } | {
-  readonly _tag: "Running"
-  readonly fiber: Fiber.RuntimeFiber<E, void>
-  readonly interruptingRef: Ref.Ref<boolean>
+class FiberStore<R, E, A> {
```

Comment:
Tbh, something like this could possibly live in an `@effect/interop`, as a general abstraction for interop with many frameworks. Food for thought..

Reply:
Maybe in the future - if we find other frameworks require similar workarounds.